trieste/models/keras/architectures.py

```diff
@@ -233,3 +235,148 @@ def connect_layers(self) -> tuple[tf.Tensor, tf.Tensor]:
         output_tensor = self._gen_output_layer(hidden_tensor)
 
         return input_tensor, output_tensor
+
+
+class DropoutNetwork(KerasEnsembleNetwork):
```

Comment:
KerasEnsembleNetwork was meant to define a standard for elements of ensemble type of model, so it doesn't make sense that you inherit from this class, even if it works in practice - you are creating a class like `KerasEnsemble`, do you really need any method from `KerasEnsembleNetwork`?

Reply:
actually, what you probably should do is subclassing `tf.keras.Model`, check option 2 in https://www.tensorflow.org/api_docs/python/tf/keras/Model

I found it a bit difficult to do that for the `KerasEnsemble` (I still plan to go back to it at some point), but here it should be straightforward, so please try to do that

Reply:
Ok, if I am doing this, is it still necessary to explicitly indicate the input shape for Trieste?

Reply:
I am struggling a bit to understand what is needed for Trieste. It seems that if subclassing from tf.keras.Model all that is needed is to create a model class that implements the correct Call method with  the number of hidden layers and output layer. In that case the builder method would only need to  use the data's output dimension to build the network and keras can infer the input shape when fit is first called? Is that right?

Reply:
what is needed by trieste  are methods defined by interfaces, how exactly a model looks like is more up to specific model wrapper, e.g. in DeepEnsemble model argument needs to be KerasEnsemble (not `tf.keras.Model`) and then that wrapper can make sure the model is correctly initialized etc.

In case of subclassing from tf.keras.Model after creating an instance of the model, calling the object with some data will create the network, I think you don't need to call fit. This could be done in the init method of the model wrapper.